src/iarc7_motion/iarc_tasks/block_roomba_task.py

```diff
@@ -90,43 +69,33 @@ def __init__(self, task_request):
         self._overshoot = (_roomba_diameter + _drone_width)/2
         self._state = BlockRoombaTaskState.init
 
-    def _receive_roomba_status(self, data):
-        with self._lock:
-            self._roomba_array = data
-
-    def _receive_landing_status(self, data):
-        with self._lock:
-            self._landed_message = data
-
-    def _current_velocity_callback(self, data):
-        with self._lock:
-            self._drone_odometry = data
-
     def get_desired_command(self):
         with self._lock:
             if self._canceled:
                 return (TaskCanceled(),)
 
             if (self._state == BlockRoombaTaskState.init):
-                if self._roomba_array is None or self._drone_odometry is None:
+                if (not self.topic_buffer.has_roomba_message()
+                 or not self.topic_buffer.has_odometry_message()):
                     self._state = BlockRoombaTaskState.waiting
                 else:
                     self._state = BlockRoombaTaskState.descent
 
             if (self._state == BlockRoombaTaskState.waiting):
-                if self._roomba_array is None or self._drone_odometry is None:
+                if (not self.topic_buffer.has_roomba_message()
```

Comment:
Since we also depend on a land message, shouldn’t we block on that here as well?